src/specs/convex/readme.md

```diff
@@ -0,0 +1,89 @@
+
+RCT exists to enable Rollup Processor to properly perform deposits and withdrawals for different users at different times. Rollup Processor expects to work with at least one input and one output asset and both assets need to be able to be owned by it. Because minted Convex LP tokens are only indirectly owned by the bridge and the actual ownership is given to crvRewards contract, Convex LP token cannot be transferred back to the Rollup Provider. RCT tokens represents the minted Convex LP tokens, but this time the token is fully owned by the bridge and can be transferred elsewhere.
+
+Gas cost of a deposit varies based on the pool. Costs mostly fluctuate in range 550k to 750k. This does not includ the transfers to/from the Rollup Processor.
+
+**Edge cases**:
+
+- If pool is not loaded, deposit to this pool will fail
+- Convex Finance has to support the Curve LP token user is trying to stake. If no pool matches the provided Curve LP token, no staking will happen.
+- Staking amount cannot be zero.
+- Pool can be shut down and no staking will happen
+
+### Withdrawal (Unstake)
+
+TL;DR: Rollup Processor transfers RCT tokens to the bridge. Bridge withdraws staked Curve LP tokens and staking rewards via Booster and direct interaction with crvRewards contract. Booster burns Convex LP tokens and returns staked Curve LP tokens to the bridge. RCT tokens are burned and returned Curve LP tokens are recovered by Rollup Processor.
+
+Before user starts with withdrawal, it is necessary to ensure that the pool for the given assets has already been loaded.
+
+Bridge expects RCT token on input and Curve LP token on output to perform withdrawal. Rollup Processor transfers RCT tokens to the bridge. Then a check is performed whether the pool for the given assets has already been loaded and the RCT token has already been deployed. Next check assures that the two assets belongs to the same pool. If both checks pass, the bridge searches for the right pool and then continues with withdrawal of Curve LP tokens. Withdrawal process consists of two steps. First, the bridge calls crvRewards contract to transfer the Convex LP tokens to the bridge and claim rewards. Second, the bridge calls the Booster to withdraw the deposited Curve LP tokens. It starts by burning the Convex LP tokens, then retrieves Curve LP tokens from Staker contract. If sufficient funds are not available, Staker pulls more deposited Curve LP tokens from the specific Curve pool liquidity gauge contract. Desired amount of Curve LP tokens is then retrieved and transferred back to the bridge. RCT tokens are burned to equally match the new amount of Convex LP tokens. Finally, Rollup Processor recovers the Curve LP tokens.
+
+Claimed rewards are the boosted CRV tokens, CVX tokens and for some pools some additional rewards. These rewards are minted for the bridge and are not recovered by the Rollup Processor.
```

Comment:
There can be 0 to many rewards. Typically rewards are CVX token and CRV token. Some pools can have some extra rewards. Once I ran a test and only like 1/6 - 1/5 of all bridges had some extra rewards. Highest number I found was 3 if I remember correctly. I did try to figure out how to claim these extra rewards but I got lost in the process. It became surprisingly complicated and they also were some no name tokens.

Reply:
Ok, please research which rewards it would even make sense to claim. If those extra rewards have tiny APR then it doesn't make sense to deal with that. But the CRV and CVX rewards should always get to the user. If the user doesn't get rewards out of the bridge then it's not very useful. When I checked matcha.xyz and 1inch they use curve to swap cvx and crv. For this reason I would dump both of these tokens to ETH and return them to the users as `outputAssetB`. Check out out curvesteth bridge for how to use curve to swap. It's pretty straightforward.
<img width="391" alt="image" src="https://user-images.githubusercontent.com/13470840/204856595-f3ded5bd-c0d6-49c7-b51e-1512592cdf2d.png">